src/Propel/Generator/Model/ForeignKey.php

```diff
@@ -625,9 +640,8 @@ public function getColumnObjectsMapping()
      */
     public function getMappedForeignColumn($local)
     {
-        $m = $this->getLocalForeignMapping();
-
-        return isset($m[$local]) ? $m[$local] : null;
+        $index = array_search($local, $this->localColumns);
+        return $this->foreignColumns[$index];
```

Comment:
No, if it can't be found, it's a internal error and should trigger warnings/errors. returning null would be wrong.